crates/router/src/core/webhooks/webhook_events.rs

```diff
@@ -325,3 +338,33 @@ async fn get_account_and_key_store(
         }
     }
 }
+
+async fn validate_event_type(
+    event_class: &HashSet<EventClass>,
+    event_type: &mut HashSet<EventType>,
+) -> Result<HashSet<EventType>, Report<errors::ApiErrorResponse>> {
```

Comment:
Nit:

```suggestion
) -> CustomResult<HashSet<EventType>, errors::ApiErrorResponse> {
```

And remove the `Report` import.